File: frontend/src/app/features/connections/models/connection-query-parameters.ts
import { QueryParameters } from "../../../core/services/rest-api/paging/query-parameters";

export interface ConnectionQueryParameters extends QueryParameters {
  withClientGaia?: string;
  withAppId?: string;
  establishedBefore?: string;
  establishedAfter?: string;
  terminatedBefore?: string;
  terminatedAfter?: string;
}

File: http-client/MCS.WatchTower.WebApi.DataTransferObjects/QueryParameters/ConnectionQueryParameters.cs
#nullable enable
namespace MCS.WatchTower.WebApi.DataTransferObjects.QueryParameters;

public record ConnectionQueryParameters(
    string? WithClientGaia = null,
    string? WithAppId = null,
    DateTime? EstablishedBefore = null,
    DateTime? EstablishedAfter = null,
    DateTime? TerminatedBefore = null,
    DateTime? TerminatedAfter = null,
    string? SearchTerm = null,
    string? OrderBy = null,
    int? Page = null,
    int? PageSize = null
) : Paging.QueryParameters(SearchTerm, OrderBy, Page, PageSize);
